src/NUnitEngine/nunit-agent/Program.cs

```diff
@@ -89,6 +89,22 @@ public static int Main(string[] args)
                 Environment.Version,
                 RuntimeFramework.CurrentFramework.DisplayName);
 
+            // Restore the COMPLUS_Version env variable if it's been overridden by TestAgency::LaunchAgentProcess
+            try
+            {
+              string cpvOriginal = Environment.GetEnvironmentVariable("TestAgency_COMPLUS_Version_Original");
+              if(!string.IsNullOrEmpty(cpvOriginal))
+              {
+                log.Debug("Agent process has the COMPLUS_Version environment variable value \"{0}\" overridden with \"{1}\", restoring the original value.", cpvOriginal, Environment.GetEnvironmentVariable("COMPLUS_Version"));
```

Comment:
I wonder if we should warn if the `TestAgency_COMPLUS_Version_Original` variable hasn't been set rather than when it has? Restoring it should be the normal case.